library/core/src/alloc/mod.rs

```diff
@@ -348,17 +348,17 @@ where
     A: AllocRef + ?Sized,
```

Comment:
`AllocRef` should be changed to be implemented on `&A` instead of `&mut A`.

Reply:
`AllocRef::by_ref` should also return `&Self`.

Reply:
done !